numpy/f2py/tests/test_regression.py

```diff
@@ -22,6 +22,25 @@ def test_inout(self):
         assert np.allclose(x, [3, 1, 2])
 
 
+class TestNegativeBounds(util.F2PyTest):
+    # Check that negative bounds work correctly
+    sources = [util.getpath("tests", "src", "negative_bounds", "issue_20853.f90")]
+
+    @pytest.mark.slow
+    def test_negbound(self):
+        xvec = np.arange(12)
+        xlow = -6
+        xhigh = 4
+        # Calculate the upper bound,
+        # Keeping the 1 index in mind
+        def ubound(xl, xh):
+            return abs(xl) + abs(xh) + 1
```

Comment:
Nit: for exact upper bound, use
```suggestion
            return xh - xl + 1
```
Notice that when using all negative bounds, say,
```
xlow = -8
xhigh = -2
```
then the abs-approach gives a wrong result for the upper bound.